store/tikv/region_request.go

```diff
@@ -146,27 +113,19 @@ func (s *RegionRequestSender) sendKVReqToRegion(bo *Backoffer, ctx *RPCContext,
 	return
 }
 
-func (s *RegionRequestSender) sendCopReqToRegion(bo *Backoffer, ctx *RPCContext, req *coprocessor.Request, timeout time.Duration) (resp *coprocessor.Response, retry bool, err error) {
-	req.Context = ctx.KVCtx
-	resp, err = s.client.SendCopReq(bo.ctx, ctx.Addr, req, timeout)
-	if err != nil {
-		if e := s.onSendFail(bo, ctx, err); e != nil {
-			return nil, false, errors.Trace(err)
-		}
-		return nil, true, nil
+func (s *RegionRequestSender) onSendFail(bo *Backoffer, ctx *RPCContext, err error) error {
+	// If it failed because the context is canceled, don't retry on this error.
+	if errors.Cause(err) == goctx.Canceled || grpc.Code(err) == codes.Canceled {
```

Comment:
Add a TODO here.
error code maybe not Canceled, and test case also need update.